metadata-ingestion/src/datahub/ingestion/source/dynamodb/dynamodb.py

```diff
@@ -480,3 +492,20 @@ def get_field_type(
 
     def get_report(self) -> DynamoDBSourceReport:
         return self.report
+
+    def _gen_domain_urn(self, dataset_name: str) -> Optional[str]:
+        for domain, pattern in self.config.domain.items():
+            if pattern.allowed(dataset_name):
+                return make_domain_urn(domain)
+
+        return None
+
+    def _get_domain_wu(
+        self, dataset_name: str, entity_urn: str
+    ) -> Iterable[MetadataWorkUnit]:
+        domain_urn = self._gen_domain_urn(dataset_name)
+        if domain_urn:
+            yield from add_domain_to_entity_wu(
```

Comment:
we have a "DomainRegistry" helper, which lets us convert domain names into urns using a lookup against DataHub 

Let's try to reuse that if possible. A good example to reference is the kafka integration's domain support: https://github.com/datahub-project/datahub/blob/1741c07d769f56a9cf066172725384b4e8780839/metadata-ingestion/src/datahub/ingestion/source/kafka.py#L445

Reply:
Is it better @hsheth2 ?

Reply:
yup this looks great!